pkgs/development/libraries/libgtkflow3/default.nix

```diff
@@ -0,0 +1,42 @@
+{stdenv, lib, vala, meson, ninja, pkg-config, fetchFromGitea, gobject-introspection, glib, gtk3, libgflow}:
+
+stdenv.mkDerivation rec {
+  pname = "libgtkflow3";
+  version = "1.0.1";
+
+  src = fetchFromGitea {
+    domain = "notabug.org";
+    owner = "grindhold";
+    repo = "libgtkflow";
+    rev = "gtkflow3_${version}";
+    hash = "sha256-9KSfDCqVSwBB+r8+4r86Kd4O6h9TvBx6SZHaAU1aBcA=";
+  };
+
+  nativeBuildInputs = [
+    vala
+    meson
+    ninja
+    pkg-config
+    gobject-introspection
+  ];
+
+  buildInputs = [
+    gtk3
+    glib
+    libgflow
```

Comment:
It does not look to me like it supports linking gflow from a different package. That would require at least changing these two lines to `dependencencies: [dependency('libgflow')]`:

https://notabug.org/grindhold/libgtkflow/src/11773a2fd02dc6d2d2ca6e9e7fdc349d7b0a0548/libgtkflow3/meson.build#L18-L19

But there are other references to the in-tree directory as well:

https://notabug.org/grindhold/libgtkflow/src/11773a2fd02dc6d2d2ca6e9e7fdc349d7b0a0548/libgtkflow3/meson.build#L27

It is also very weird that they use `custom_target` over [`gnome.generate_gir`](https://mesonbuild.com/Gnome-module.html#gnomegenerate_gir).

So until we manage to make it build with `postPatch = ''rm -r libgflow''`, we probably should not it split like this.

So we either need to patch the code to use external libgflow (similarly like we do with [gssdp here](https://github.com/NixOS/nixpkgs/blob/c1487a18d133ff8c15b674130c618c128ba7b4fe/pkgs/development/libraries/gssdp/standalone-tools.patch)), or give up on perfect split (like we do in [libportal](https://github.com/NixOS/nixpkgs/blob/c1487a18d133ff8c15b674130c618c128ba7b4fe/pkgs/development/libraries/libportal/default.nix), where both `libportal-gtk3` and `libportal-gtk4` duplicate the `libportal` library.)